Core/DISET/private/BaseClient.py

```diff
@@ -304,57 +420,101 @@ def _connect( self ):
         if not result['OK']:
           return result
 
+    # Take all the extra credentials
     self.__discoverExtraCredentials()
     if not self.__initStatus[ 'OK' ]:
       return self.__initStatus
     if self.__enableThreadCheck:
       self.__checkThreadID()
+
     gLogger.debug( "Connecting to: %s" % self.serviceURL )
     try:
+      # Calls the transport method of the apropriate protocol.
+      # self.__URLTuple[1:3] = [server name, port, System/Component]
       transport = gProtocolDict[ self.__URLTuple[0] ][ 'transport' ]( self.__URLTuple[1:3], **self.kwargs )
       #the socket timeout is the default value which is 1.
       #later we increase to 5
       retVal = transport.initAsClient()
+      # If we have an issue connecting
       if not retVal[ 'OK' ]:
+        # We try at most __nbOfRetry each URLs
         if self.__retry < self.__nbOfRetry * self.__nbOfUrls - 1:
+          # Recompose the URL (why not using self.serviceURL ? )
           url = "%s://%s:%d/%s" % ( self.__URLTuple[0], self.__URLTuple[1], int( self.__URLTuple[2] ), self.__URLTuple[3] )
+          # Add the url to the list of banned URLs if it is not already there. (Can it happen ? I don't think so)
           if url not in self.__bannedUrls:
             self.__bannedUrls += [url]
+            # Why only printing in this case ?
             if len( self.__bannedUrls ) < self.__nbOfUrls:
               gLogger.notice( "Non-responding URL temporarily banned", "%s" % url )
+          # Increment the retry couunter
           self.__retry += 1
+          # If it is our last attempt for each URL, we increase the timeout
           if self.__retryCounter == self.__nbOfRetry - 1:
             transport.setSocketTimeout( 5 ) # we increase the socket timeout in case the network is not good
           gLogger.info( "Retry connection: ", "%d" % self.__retry )
+          # If we tried all the URL, we increase the global counter (__retryCounter), and sleep
           if len(self.__bannedUrls) == self.__nbOfUrls:
             self.__retryCounter += 1
             self.__retryDelay = 3. / self.__nbOfUrls  if self.__nbOfUrls > 1 else 2  # we run only one service! In that case we increase the retry delay.
             gLogger.info( "Waiting %f seconds before retry all service(s)" % self.__retryDelay )
             time.sleep( self.__retryDelay )
+          # rediscover the URL
           self.__discoverURL()
+          # try to reconnect
           return self._connect()
         else:
           return retVal
     except Exception as e:
       gLogger.exception(lException = True, lExcInfo = True)
       return S_ERROR( "Can't connect to %s: %s" % ( self.serviceURL, repr( e ) ) )
+    # We add the connection to the transport pool
     trid = getGlobalTransportPool().add( transport )
+
     return S_OK( ( trid, transport ) )
 
   def _disconnect( self, trid ):
+    """ Disconnect the connection.
+        :param trid: Transport ID in the transportPool
```

Comment:
Can't we have a post commit or post push hook to verify that kind of things ? Or even a test travis style ?